include/magma/vk/ShaderModule.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include <vulkan/vulkan.hpp>
+#include <string>
+#include <vector>
+
+enum ShaderStage {
```

Comment:
it's better to make this "enum class" instead of "enum".
Also it's better to be an internal enum of shader class